tests/helper/utils.py

```diff
@@ -72,6 +72,42 @@ def get_file_perm(client, fname):
     if not "cannot statx" in error:
         return int(output)
 
+def get_installed_kernel_versions(client):
+    """Get a list of installed kernel versions using 'linux-version list'."""
+    command = "linux-version list"
+    (exit_code, output, error) = client.execute_command(command, quiet=True)
+    if exit_code == 0 and output:
+        kernel_versions = output.strip().split('\n')
+        return kernel_versions
+    else:
+        return []
+
+def check_kernel_module_exists_for_kernel_version(client, module_name, kernel_version):
+    """ Checks if a kernel module exists for a given kernel version """
+    
+    for ext in ["ko", "ko.gz", "ko.xz", "ko.bz2", "ko.zst"]:
+        filename = f"{module_name}.{ext}"
+        command = f"find /lib/modules/{kernel_version} -type f -name {filename}"
+        (exit_code, output, error) = client.execute_command(command, quiet=True)
+        if exit_code == 0 and output:
+            return True
+
+    return False
```

Comment:
I'm missing experience with pytest, so for me it's not clear here if this fails the test or not. Should it fail the test? If so is there a more self-describing way like raising a new exception?

Reply:
This is the helper function for the actual pytest (with a proper assertion)

features/cloud/test/test_kernel_modules.py:
```
def test_kernel_modules(client, module_name):
     """Check if the module_name kernel module is available as loadable module """
     # NOTE: check all kernel versions, in case multiple kernels are installed
     assert utils.check_module_exists_in_all_kernels(client, module_name), f"{module_name} does not exist in image"
```

Reply:
We could also raise an exception in that helper function true.

 I suggest to keep the style coherent with the other helper functions for now. Not sure if a proper exception handling is beneficial or more elegant.

Reply:
thanks, makes sense